packages/storybook/src/documentation/pages/demos/pagination/serverPagination.demo.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { Columns, FeatureMode, PageChangedParams, RowsProp, XGrid } from '@material-ui/x-grid';
+import {
+  randomCreatedDate,
+  randomEmail,
+  randomId,
+  randomInt,
+  randomTraderName,
+  randomUpdatedDate,
+} from '@material-ui/x-grid-data-generator';
+
+function newRow() {
+  return {
+    id: randomId(),
+    name: randomTraderName(),
+    email: randomEmail(),
+    age: randomInt(10, 100),
+    dateCreated: randomCreatedDate(),
+    lastLogin: randomUpdatedDate(),
+  };
+}
+
+const columns: Columns = [
+  { field: 'id', hide: true },
+  { field: 'name', type: 'string' },
+  { field: 'email', type: 'string' },
+  { field: 'age', type: 'number' },
+  { field: 'dateCreated', type: 'date', width: 180 },
+  { field: 'lastLogin', type: 'dateTime', width: 180 },
+];
+
+function loadServerRows(params: PageChangedParams): Promise<any> {
+  return new Promise<any>((resolve) => {
+    const rows: any[] = [];
+    while (rows.length < params.pageSize) {
+      rows.push(newRow());
+    }
+
+    setTimeout(() => {
+      resolve(rows);
+    }, 800);
+  });
+}
+
+export default function ServerPaginationDemo() {
+  const [rows, setRows] = React.useState<RowsProp>([]);
+  const [isLoading, setLoading] = React.useState<boolean>(false);
+
+  const onPageChanged = (params) => {
+    setLoading(true);
+    loadServerRows(params).then((newRows) => {
+      setRows(newRows);
```

Comment:
Need to prevent setState if the component unmounts.

> So how do you want to do that? Are you sure you want to add this complexity in the demo?

Yes, we especially want to do it if it adds complexity, it's great to feel the pain around it: "here is what it's required to do it right: ok, it's hard, ok maybe we can improve the API to make it easier".

In any case, it's important to showcase best practices.

Reply:
We have a couple of options, linking them:

Reply:
- active variable: https://github.com/mui-org/material-ui/blob/b2c98cfd4146a8dfcfc4d12fbf6f77a2146bff1a/docs/src/pages/components/autocomplete/GoogleMaps.tsx#L97
- cancel request: https://github.com/mui-org/material-ui-pickers/blob/dfbbe65d7ba4f858a2e660c19d49ab12f5e4fe87/docs/pages/demo/datepicker/ServerRequest.example.jsx#L15

Reply:
Actually, outside of the unmounting issue, there is another one that is even more important that I forgot about: concurrent resolution. If users switch from page 1 to page 2 then quickly to page 3 but the request for page 2 resolve after page 3, we still need to make sure we only consider the data of page 3, not the latest 2.

Reply:
Actually, I'm confused, for server-side pagination, won't we need a different API? I don't see how this once can scale to grouping, filtering, sorting. Maybe we don't need the server-side pagination, but only a server-side data fetcher (that would solve the unmounting and concurrency issues directly.

Reply:
Well of course, concurrency is not managed in the demo as this should be managed by the consumer.
I can introduce Rx and do a switch map but it's a demo and I thought we wanted to keep them as simple as possible

Reply:
What about we persiste the active page on a React state and use an effect on the active page to fetch new data? By combining it with a `active ` variable, we would keep the demo simple, be resilient, and avoid the usage of the API ref for simple cases.

Reply:
Regarding Rx.js, as far as I know, it's rare to see it used in the React community. We could have a demo for it, but I think that it should cover the advanced cases that **can't** be done with hooks or the declarative API otherwise.

Reply:
There is no usage of apiRef in this demo. 
Yes or we can add a correlation id or check if the page is still the same, but I don't think it will bring any value to the demo. We just want to show how the grid works and not show how to manage concurrency in a web app.
Concurrency is outside the scope of the grid component. 

If the component would handle the request, then we would need to manage concurrency. ATM we don't.

Reply:
Well RxJs is massively popular in react as well. Angular is built on it to manage events...
In react, events are managed by react, and server requests are out of the react scope. 

That being said, in a react app it's up to you, how you build your communication layer with the server. You can use axios, fetch or rxjs if you want.
This is what makes react great and this is also why ppl go with angular as they want a full layer architecture and not just the component or rendering layer

Reply:
The proposed approach would fix two bugs with the current demo that people will face as soon as they copy the demo in their app and replace the fetch with a real network request.

Will it make the demo more complex? Not sure

Reply:
Ok I do it

Reply:
Done

Reply:
Do you want to manage server errors in this demo as well? Maybe add a catch with a server boundary, or we can add an error overlay in the grid which I think would be great